{% include 'header.html' %}


    <!-- Fixed navbar -->
    <nav class="navbar navbar-inverse navbar-fixed-top">
      <div class="container">
        <div class="navbar-header">
          <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
          </button>
          <a class="navbar-brand" href="/">plant.MAP</a>
        </div>
        <div id="navbar" class="navbar-collapse collapse">
          <ul class="nav navbar-nav">
            <li><a href="/">Home</a></li>
            <li class="active"><a href="#about">About</a></li>
            <li><a href="download">Download</a></li>
          </ul>
        </div>
      </div>
    </nav>

{% include 'jumbotron.html' %}


  <div class="container theme-showcase" role="main">
      <div class="page-header">
        <strong><h3>About</h3></strong>
      </div>
      <p>
      Proteins interact with each other and organize themselves into macromolecular machines (ie. complexes) to carry out essential functions of the cell. 
      </p>
  
      <ul>
   
      <h5>Protein interactions</h5>
    
      <li>We derived protein interactions from co-fractionation mass spectrometry of 12 taxonomically diverse plant species</li>
      <li>These are stable protein interactions conserved across plants</li>
      <li>Interactions can be queried using either UniProt Entry, Entry name, or TAIR AT number.</li>
      <li>False discovery rates of protein interactions above CF-MS score 0.5 are 10%, and above CF-MS score 0.2 are 50%</li>
      <li>Interactions with CF-MS scores > 0.5 are descriptive of stable interactions, but a score above 0.2 is commonly descriptive of a more transient protein interactions</li>
      </ul>
 
      <h5>Protein complexes</h5>
      <ul>
      <li>Protein complexes were derived from interactions that pass high thresholds</li>
      <li>Protein complexes are presented with four different levels of clustering stringency, from most restrictive to most inclusive</li>
      </ul>
  
      <h5>Orthology</h5>
      <ul>
      <li>As protein complexes were derived from multi-species data, they are indexed by <a href = "http://eggnogdb.embl.de">eggNOG DB</a> plant-level orthogroupID</li>
      <li>These orthogroups take the form ENOG411XXXX, and can be searched at the <a href = "http://eggnogdb.embl.de">eggNOG website</a></li>
      <li>Due to gene expansions since the last common ancestor of plant, each orthogroup may contain more than one protein from a species</li>
      </ul> 

      <br>For further details see paper <li><a href="" target="blank">PAPER LINK </a></li>

      <h4>Citation</h4>
      <ul>
          <li><a href="" target="blank">
          PAPER CITATION   
      </a></li>
      </ul>


      <br>
      <h4>Website contributers</h4>
      <p>
      <strong>&nbsp Claire McWhite</strong>(<a href="http://clairemcwhite.github.io/" target="_blank">website</a>)
      <br>
      &nbsp Kevin Drew (<a href="http://ksdrew.github.io/" target="_blank">website</a>)
      <br>
      &nbsp Edward Marcotte (<a href="http://marcottelab.org" target="_blank">website</a>)
      <br>

      </p>

      <br>

      <h4>Contact</h4>
      &nbsp clairemcwhite (at) utexas.edu
      <br>

{% include 'footer.html' %}

