{% include 'header.html' %}
    <!-- Fixed navbar -->
    <nav class="navbar navbar-inverse navbar-fixed-top">
      <div class="container">
        <div class="navbar-header">
          <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
          </button>
          <a class="navbar-brand" href="/">plant.MAP</a>
        </div>
        <div id="navbar" class="navbar-collapse collapse">
          <ul class="nav navbar-nav">
            <li><a href="/">Home</a></li>
            <li><a href="/about">About</a></li>
            <li class="active"><a href="/download">Download</a></li>
          </ul>
        </div><!--/.nav-collapse -->
      </div>
    </nav>

    <div class="container theme-showcase" role="main">
{% include 'jumbotron.html' %}

      <div class="page-header">
        <h3>Download</h3>
      </div>
      <h4>Plant Complex Map Files</h4>
      <ul>

          <li><a href="/static/data/panplant_annotated_clusters.xlsx" download target="_blank">Protein Complexes (Excel format, with annotations and protein elution profiles)</a>
              <ul>
                  <li>Description: Annotated complexes generated from walktrap diffusion clustering of FDR < 10% interactions (CF-MS score cutoff 0.509) </li>
              </ul>                                                                            
          </li>
          <li><a href="/static/data/panplant_clusters.csv" download target="_blank">Protein Complexes (Text format, no annotations or elution profiles)</a>
              <ul>
                  <li>Description: Plain text complexes generated from walktrap diffusion clustering of FDR < 10% interactions (CF-MS score cutoff 0.509) </li>
                 <li>Note: The leftmost columns are more inclusive (larger) clusters, rightmost are more restrictive (smaller) </li>
                 <li>More restrictive clustering tends to separate complexes into subcomplexes</li>
                 <li>Cluster numbers give guidance as to which proteins are likely to belong to the same complex. Note that the numbers are only informative within the same column.</li>
              </ul>                                                                            
          </li>

          <li><a href="/static/data/panplant_cfms_scores_annot.txt.gz" download target="_blank">All scored pairwise protein-protein interactions with CF-MS scores (3,076,999 unique pairwise interactions)</a>
              <ul>
                  <li>Description: Scores between Orthogroups with the corresponding CF-MS score and eggNOG generated orthogroup descriptions. </li>
                  <li>Note: Only the highest scoring pairs are considered significant. A CF-MS score >= 0.509 corresponds to 10% FDR, >= 0.207 corresponds to 50% FDR</li>
                  <li>Format: OrthogroupID1 [tab] OrthogroupID2 [tab] Score [tab] Annotation1 [tab] Annotation2</li>
              </ul>
          </li>
      </ul>
      <br>
      <h4>Additional database input files</h4>
      <ul>
          <li><a href="/static/data/virNOG_collapse_annotations.txt" download target="_blank">Orthogroup Annotations</a>
              <ul>
                  <li>Description: Annotations for all orthogroups</li>
              </ul>
          </li>
          <li><a href="/static/data/plant_virNOG_orthology.csv" download target="_blank">Translation Table (i.e., the precise orthogroup definitions used for this analysis)</a>
              <ul>
                  <li>Mapping of eggNOG virNOG OrthogroupID to ProteinIDs used in bioinformatics pipeline</li>
                  <li>Generated with EggNOG mapper against EggNOG4.0 orthogroups</li>
                  <li>Format: OrthogroupID1 [tab] ProteinID [tab] Species code </li>       
              </ul>
          </li>
          <li><a href="/static/data/complete_orthogroup_protID_mapping.csv" download target="_blank">Translation Table (contains additional mappings used for web site queries)</a>
              <ul>
                  <li>Mapping of eggNOG virNOG OrthogroupID to different encodings of ProteinID</li>
                  <li>TAIR AT locus, uniprot Accession, uniprot Entry name, Rice locus, Arabidopsis gene name</li>
                  <li>Format: OrthogroupID1 [tab] ProteinID [tab] Species code [tab] Identifier type</li>       
              </ul>
          </li>

      </ul>
      <br>

      <h4>Metadata</h4>
      <ul>
          <li><a href="/static/data/Experiment_Order.csv" download target="_blank">Experiment metadata</a>
              <ul>
                  <li>Description of each fractionation experiment </li>
                  <li>Format: experiment_name,ExperimentID_order,tissue,experiment_type,spec,ExperimentID</li>
              </ul>
          </li>
          <li><a href="/static/data/Fraction_Details.csv" download target="_blank">Fraction metadata</a>
              <ul>
                  <li>Description of each fraction in all fractionation experiment </li>
                  <li>FractionID,frac_order,ExperimentID</li>
              </ul>
          </li>
      </ul>   


      <h4>Test and training interactions</h4>
      Identifiers (OrthogroupIDs) are <a href="http://eggnogdb.embl.de/">eggNOG</a> <i>viridiplantae</i> orthogroup identifiers
      <ul>
          <li><a href="/static/data/panplant_pos_train_ppis.txt" download target="_blank">Train Positive PPIs</a>
              <ul>
                  <li>Description: List of training positive ppis used in protein complex discovery pipeline</li>
                  <li>Contain PPI from CORUM human protein complexes transfered to plant ortholog groups, as well as other supported PPI from TAIR</li>
                  <li>Format: OrthogroupID [tab] OrthogroupID </li>
              </ul>
          </li>
          <li><a href="/static/data/panplant_neg_train_ppis.txt" download target="_blank">Train Negative PPIs</a>
              <ul>
                  <li>Description: List of train negative ppis used in protein complex discovery pipeline</li>
                  <li>Note: Since no truly non-interacting proteins exist, these are randomly sampled from observed proteins in our data set</li>
                  <li>Format: OrthogroupID [tab] OrthogroupID </li>
              </ul>
          </li>
          <li><a href="/static/data/panplant_pos_test_ppis.txt" download target="_blank">Test Positive PPIs</a>
              <ul>
                  <li>Description: List of test positive ppis used in protein complex discovery pipeline</li>
                  <li>Format: OrthogroupID [tab] OrthogroupID </li>
              </ul>
          </li>
          <li><a href="/static/data/panplant_neg_test_ppis.txt" download target="_blank">Test Negative PPIs</a>
              <ul>
                  <li>Description: List of test negative ppis used in protein complex discovery pipeline</li>
                  <li>Note: Since no truly non-interacting proteins exist, these are randomly sampled from observed proteinsin our data set</li>
                  <li>Format: OrthogroupID [tab] OrthogroupID </li>
              </ul>
          </li>
      </ul>
      <br>
      <h4>Elution table</h4>
      <ul>
          <li><a href="/static/data/panplant_wide_elution_expnorm.csv" download target="_blank">Concatenated elution profiles (125 MB)</a>
              <ul>
                  <li>Description: Table of concatenated elution profiles reported per-fractionation-normalized peptide spectral matches (PSMs) for all observed orthogroups </li>
                  <li>Format: OrthogroupID,[Fractions]</li>
              </ul>
          </li>
          <li><a href="/static/data/panplant_tidy_elution.csv" download target="_blank">Tidy elution profile (455MB)</a>
              <ul>
                  <li>Description: Tidy format concatenated elution profiles of all orthogroups </li>
                  <li>Includes a variety of abundance measurements and normalizations</li> 
                  <li>fracnorm = normalized by each fraction</li>
                  <li>expnorm = normalized by fractionation experiment</li>
                  <li>Format: filename,ExperimentID,FractionID,ID,Total_PeptideCount,abundance_ppm,spec,fracnorm_PeptideCount,expnorm_PeptideCount,expnorm_ppm,fracexpnorm_PeptideCount</li>
              </ul>
          </li>
      </ul>   

      <h4>Feature Matrix</h4>
      <ul>
          <li><a href="/static/data/panplant_featmat_trainlabeled.csv.gz" download target="_blank">Feature Matrix (gz) (3GB)</a>
              <ul>
                  <li>Description: All calculated features with training labels from McWhite & Papoulas et al.</li>
                  <li>Format: OrthogroupID1[space]OrthogroupID2,[features],OrthogroupID1,OrthogroupID2,label</li>
              </ul>
          </li>
      </ul>
      <ul>
          <li><a href="/static/data/panplant_featmat_top100feats_trainlabeled.csv.gz" download target="_blank">Feature Matrix (gz) (1.4GB)</a>
              <ul>
                  <li>Description: Feature-selected top 100 features with training labels from McWhite & Papoulas et al.</li>
                  <li>Format: OrthogroupID1[space]OrthogroupID2,[features],label</li>
                  <li>Note: Orthogroup1,Orthogroup2 are in alphatical order</li> 
              </ul>
          </li>
      </ul>



      <br>
      <h4>Code</h4>
      <ul>
          <li>Plant Protein Complex Map <a href="http://doi.org/10.5281/zenodo.3466034" target="_blank"> Zenodo repository</a></li>
          <li>Notes: Zenodo repository contains all files, instructions to replicate, and internal scripts</li>
          <li>Project makes use of following github repositories</li>
          <li><a href="https://github.com/marcottelab/MS_grouped_lookup" target="_blank">Peptide detection by orthogroup (GitHub)</a></li>
          <li><a href="https://github.com/marcottelab/protein_complex_maps" target="_blank">Scripts for calculating interaction networks</a></li>
         
      </ul>

      <br>
      <h4> License </h4>
      <ul>
          <li><a href="https://creativecommons.org/licenses/by/4.0/" target="_blank">CC BY 4.0</a>
          </li>
          <li>
          Data associated with this website are free to download and share.  They are governed by the <a href="https://creativecommons.org/licenses/by/4.0/" target="_blank">Creative Commons Attribution 4.0 International (CC BY 4.0) license</a>, which means that they are a part of the public domain, and every use of them is allowed, provided that you give appropriate credit, provide a link to the license, and indicate if changes were made. You may do so in any reasonable manner, but not in any way that suggests the licensor endorses you or your use.  If you make extensive use of data from this data set, please credit the authors and when appropriate the authors of the source data (see <a href="/about">about</a> for references).
          </li>
      </ul>
      <br>
 {% include 'footer.html' %}
 
